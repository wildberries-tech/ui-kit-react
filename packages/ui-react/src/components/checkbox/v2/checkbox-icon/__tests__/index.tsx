import React from 'react';
import * as renderer from 'react-test-renderer';

import { CheckboxIcon } from '..';

const defaultProps = {
    checked: false,
    indeterminate: false,
    disabled: false
};

test('Default', () => {
    const tree = renderer.create(<CheckboxIcon {...defaultProps} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Is checked', () => {
    const tree = renderer.create(
        <CheckboxIcon 
            {...defaultProps} 
            checked={true}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Is indeterminate', () => {
    const tree = renderer.create(
        <CheckboxIcon 
            {...defaultProps} 
            indeterminate={true}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('Is disabled', () => {
    const tree = renderer.create(
        <CheckboxIcon 
            {...defaultProps} 
            disabled={true}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});