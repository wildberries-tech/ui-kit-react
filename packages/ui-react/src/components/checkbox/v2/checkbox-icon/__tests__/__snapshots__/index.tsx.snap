// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default 1`] = `
<svg
  className="checkbox-icon"
  height={24}
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    className="checkbox-icon__rect"
    d="M2.223 4.741a2.968 2.968 0 0 1 2.518-2.518 23.042 23.042 0 0 1 6.518 0 2.968 2.968 0 0 1 2.518 2.518 23.049 23.049 0 0 1 0 6.518 2.968 2.968 0 0 1-2.518 2.518 23.049 23.049 0 0 1-6.518 0 2.968 2.968 0 0 1-2.518-2.518 23.042 23.042 0 0 1 0-6.518Z"
  />
  <path
    className="checkbox-icon__check"
    d="m5 7.8 2.3 2.3 4.271-4.6"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</svg>
`;

exports[`Is checked 1`] = `
<svg
  className="checkbox-icon"
  height={24}
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    className="checkbox-icon__rect"
    d="M2.223 4.741a2.968 2.968 0 0 1 2.518-2.518 23.042 23.042 0 0 1 6.518 0 2.968 2.968 0 0 1 2.518 2.518 23.049 23.049 0 0 1 0 6.518 2.968 2.968 0 0 1-2.518 2.518 23.049 23.049 0 0 1-6.518 0 2.968 2.968 0 0 1-2.518-2.518 23.042 23.042 0 0 1 0-6.518Z"
  />
  <path
    className="checkbox-icon__check"
    d="m5 7.8 2.3 2.3 4.271-4.6"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</svg>
`;

exports[`Is disabled 1`] = `
<svg
  className="checkbox-icon"
  height={24}
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    className="checkbox-icon__rect"
    d="M2.223 4.741a2.968 2.968 0 0 1 2.518-2.518 23.042 23.042 0 0 1 6.518 0 2.968 2.968 0 0 1 2.518 2.518 23.049 23.049 0 0 1 0 6.518 2.968 2.968 0 0 1-2.518 2.518 23.049 23.049 0 0 1-6.518 0 2.968 2.968 0 0 1-2.518-2.518 23.042 23.042 0 0 1 0-6.518Z"
  />
  <path
    className="checkbox-icon__check"
    d="m5 7.8 2.3 2.3 4.271-4.6"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
  <path
    className="checkbox-icon__disabled"
    d="m3 3 10 10"
  />
</svg>
`;

exports[`Is indeterminate 1`] = `
<svg
  className="checkbox-icon"
  height={24}
  viewBox="0 0 16 16"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    className="checkbox-icon__rect"
    d="M2.223 4.741a2.968 2.968 0 0 1 2.518-2.518 23.042 23.042 0 0 1 6.518 0 2.968 2.968 0 0 1 2.518 2.518 23.049 23.049 0 0 1 0 6.518 2.968 2.968 0 0 1-2.518 2.518 23.049 23.049 0 0 1-6.518 0 2.968 2.968 0 0 1-2.518-2.518 23.042 23.042 0 0 1 0-6.518Z"
  />
  <rect
    className="checkbox-icon__indeterminate"
    height="2"
    rx="1"
    width="6"
    x="5"
    y="7"
  />
  <path
    className="checkbox-icon__check"
    d="m5 7.8 2.3 2.3 4.271-4.6"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
</svg>
`;
